index.js

```diff
@@ -149,8 +149,14 @@ walletRouter.get('/balance', (req, res) => {
 
 walletRouter.get('/address', (req, res) => {
   const wallet = req.wallet;
-  const markAsUsed = req.query.mark_as_used || false;
-  const address = wallet.getCurrentAddress({markAsUsed});
+  const index = req.query.index || null;
+  let address;
+  if (index) {
```

Comment:
Is `0` a valid index? Should we test for null instead implicit bool, which would put `index=0` with the same behavior as no index?

Reply:
Makes sense! Changed to `if (index !== null)`, much better